queue.go

```diff
@@ -106,12 +108,39 @@ type Queue struct {
 	workers      []worker
 }
 
-// NewQueue creates and returns a new Queue structure
-func NewQueue(url *URL, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
-	q := &Queue{
+// NewQueue returns a new Queue structure
+func NewQueue(url *URL, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Channel, error) {
+	q := &Channel{
 		name:         name,
 		url:          url.urlString(),
 		isConsumer:   isConsumer,
+		isExchange:   false,
+		Durable:      durable,
+		Jobs:         jobs,
+		prefetchSize: prefetchSize,
+		workers:      make([]worker, 0),
+	}
+
+	if err := q.connect(); err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(context.TODO())
+
+	q.cancelCtx = cancel
+
+	go q.reconnector(ctx)
+
+	return q, nil
+}
+
+// NewExchange returns a new Exchange structure
+func NewExchange(url *URL, name string, prefetchSize int, durable bool, jobs chan amqp.Delivery) (*Channel, error) {
```

Comment:
Body of `NewQueue()` and `NewExchange()` is almost identical. I suggest that you create a new `NewChannel()` method this this body that is called by `NewQueue()` and `NewExchange()` with appropriate parameters